trajectorytools/export/to_pandas.py

```diff
@@ -0,0 +1,167 @@
+import collections
+import functools
+import itertools
+from typing import Dict, List, Optional
+
+import numpy as np
+import pandas as pd
+
+from variables import get_variable
+
+
+def melt_df(df: pd.DataFrame, value_name: str, var_name: str):
+    melted_df = pd.melt(
+        df,
+        id_vars=[col for col in df.columns if isinstance(col, str)],
+        value_vars=[col for col in df.columns if not isinstance(col, str)],
+        var_name=var_name,
+        value_name=value_name,
+    )
+    return melted_df
+
+
+def generate_var_df(
+    x: np.ndarray,
+    y: np.ndarray,
+    x_name: str,
+    y_name: str,
+    identities: Optional[Dict[str, List]] = None,
+) -> pd.DataFrame:
+    assert len(x) == y.shape[0]
+    df = pd.DataFrame(
+        data=y.T, columns=list(x)
+    )  # Transpose so that columns are time steps
+    if identities is not None:
+        for key in identities:
+            df[key] = identities[key]
+    return melt_df(df, y_name, x_name)
+
+
+def get_focal_nb_ids(key: str, identities: List):
+    focal_neighbour_ids_conds = list(itertools.product(identities, identities))
+    focals, neighbours = list(zip(*focal_neighbour_ids_conds))
+    dict_vars = {
+        key: focals,
+        f"{key}_nb": neighbours,
+    }
+    return dict_vars
+
+
+def is_focal_nb_variable(tr, var_array):
+    if len(var_array.shape) == 3:
+        frames_cond = len(var_array) == tr.number_of_frames
+        num_indiv_cond = var_array.shape[1] == tr.number_of_individuals
+        focal_nb_cond = var_array.shape[1] == var_array.shape[2]
+        return frames_cond and num_indiv_cond and focal_nb_cond
+    else:
+        return False
+
+
+def is_individual_variable(tr, var_array):
+    if len(var_array.shape) == 2:
+        frames_cond = len(var_array) == tr.number_of_frames
+        num_indiv_cond = var_array.shape[1] == tr.number_of_individuals
+        return frames_cond and num_indiv_cond
+    else:
+        return False
+
+
+def is_group_variable(tr, var_array):
+    if len(var_array.shape) == 1:
+        frames_cond = len(var_array) == tr.number_of_frames
+        return frames_cond
+    else:
+        return False
+
+
+def tr_variable_to_df(tr, var):
+    identities = list(range(tr.number_of_individuals))
```

Comment:
I would advise to use the method identity_labels

Reply:
will do